### PR TITLE
Add d4j support

### PR DESCRIPTION
**Pull request by DRSchlaubi**

### CODE REVIEW

File: src/test/kotlin/d4jTestScript.kt [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+        "play" -> {
+            val input = event.getOption("identifier")
+                .flatMap(ApplicationCommandInteractionOption::getValue)
+                .map(ApplicationCommandInteractionOptionValue::asString)
+                .get()
+            val link = lavalink.getLink(event.interaction.guildId.get().asLong())
+            link.getPlayer().block()!!.setIdentifier(input)
+                .asMono()
+                .block()
```
Comment (duncte123) [inline src/test/kotlin/d4jTestScript.kt:108]: Please call subscribe instead
```suggestion
                .subscribe()
```

File: build.gradle.kts [hunk 2]
```diff
@@ -28,11 +28,13 @@ dependencies {
 
     // Discord library support
     compileOnly("net.dv8tion:JDA:5.0.0-beta.11")
+    compileOnly("com.discord4j:discord4j-core:3.2.3")
     // I have no clue how this lib works and the docs are confusing
     // compileOnly("com.discord4j:discord4j-core:3.2.5")
```
Comment (duncte123) [inline build.gradle.kts:33]: remove this pls

**Review by topi314 — APPROVED** [review]

**Comment by topi314** [comment]
lgtm

File: build.gradle.kts (hunk 2 above)
Comment (duncte123) [inline build.gradle.kts:33]:
```suggestion
```

**Review by duncte123 — APPROVED** [review]

**Pull request merged**